pkg/repl/repl.go

```diff
@@ -0,0 +1,71 @@
+// Package repl is a simple read-eval-print loop.  It calls the Consumer
+// to do all the eval work.
+package repl
+
+import (
+	"io"
+
+	"github.com/peterh/liner"
+)
+
+type Consumer interface {
+	Consume(line string) bool
+	Prompt() string
+}
+
+type REPL struct {
+	consumer Consumer
+}
+
+func NewREPL(c Consumer) *REPL {
+	return &REPL{consumer: c}
+}
+
+// Run executes the REPL.
+func (r *REPL) Run() error {
+	l := liner.NewLiner()
+	defer l.Close() //nolint:errcheck
+	l.SetMultiLineMode(true)
+	for {
+		line, e := l.Prompt(r.consumer.Prompt())
+		if e == io.EOF {
+			return io.EOF
+		} else if e != nil {
+			// Ignore this error; the prior is more interesting.
```

Comment:
I don't understand this comment.

Reply:
I don't get it either. In fact this whole stanza could be simplified.